memiavl/db.go

```diff
@@ -97,13 +100,24 @@ func (db *DB) Commit(changeSets MultiChangeSet) ([]byte, int64, error) {
 			}
 
 			// snapshot rewrite succeeded, catchup and switch
-			// TODO prune the old snapshots
 			if err := result.mtree.CatchupWAL(db.wal); err != nil {
 				return nil, 0, fmt.Errorf("catchup failed: %w", err)
 			}
 			if err := db.reloadMultiTree(result.mtree); err != nil {
 				return nil, 0, fmt.Errorf("switch multitree failed: %w", err)
 			}
+			// prune the old snapshots
+			version := uint32(db.lastCommitInfo.Version)
+			latestSnapshot := snapshotName(version)
+			entries, err := os.ReadDir(db.dir)
+			if err == nil {
+				for _, entry := range entries {
+					if entry.IsDir() && strings.HasPrefix(entry.Name(), SnapshotPrefix) &&
+						entry.Name() != latestSnapshot {
+						os.RemoveAll(filepath.Join(db.dir, entry.Name()))
```

Comment:
Handle the error return?

Reply:
I think can ignore the error here actually

Reply:
make sense, in the future we should add an error log then.
And I think it makes more sense to do the pruning in a background go-routine.